### ISSUE TITLE
Question about inference on test images.(Memory out)

### ISSUE THREAD

**Issue by KeuntekLee** [liuzhen03/ADNet#4]
Hi. 
Thanks for your works. BTW, i have question about GPU memory on test time.

I read your paper, especially about test evaluation.

> The entire testing process is conducted on a single
2080Ti GPU. Constrained by the limited GPU memory, we
split each test image into size 1060x1000 and 1060x900
for testing and then concatenate them to the full size
1060x190

However, when i test on my single RTX 2080Ti GPU(12G VRAM), i faced CUDA memory out even with single PCD alignment module.
My testing code is like below.
```import os
os.environ["CUDA_VISIBLE_DEVICES"] = "3"
import torch
from dcn_v2 import PCD_Align
def requires_grad(model, flag=True):
    for p in model.parameters():
        p.requires_grad = flag
a = torch.randn([1,64,1060,900]).cuda()
b = torch.randn([1,64,530,450]).cuda()
c = torch.randn([1,64,265,225]).cuda()
model = PCD_Align().cuda()
requires_grad(model,False)

out = model([a,b,c], [a,b,c])
```

As you described on paper, i made tensors which have same shape of after through Pyramid module(assume original LDR image size is 1060x900). And also turned off gradient of network for saving memory.

Even i pass tensors only 1 PCD Alignment(need to run 2 times in full inference), it caused OOM on CUDA.
To make sure, i attach my GPU status with !nvidia-smi command

>Thu Aug 12 22:37:59 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.80       Driver Version: 460.80       CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  Off  | 00000000:24:00.0 Off |                  N/A |
| 87%   80C    P2   234W / 250W |   5900MiB / 11019MiB |     98%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 208...  Off  | 00000000:2F:00.0 Off |                  N/A |
| 50%   48C    P8    16W / 250W |   7372MiB / 11016MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  GeForce RTX 208...  Off  | 00000000:30:00.0 Off |                  N/A |
| 99%   86C    P2   196W / 250W |  10933MiB / 11019MiB |     93%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  GeForce RTX 208...  Off  | 00000000:31:00.0 Off |                  N/A |
| 39%   49C    P8    12W / 250W |      8MiB / 11019MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

Can you answer this problem? It'll be great if you share some details on your test process.
BR.

**Comment by liuzhen03** [comment]
Hi, I have run your code as below (single 2080Ti GPU, 11G RAM):

```
def test_pcd():
    print("Test PCD Module")
    def requires_grad(model, flag=True):
        for p in model.parameters():
            p.requires_grad = flag
    a = torch.randn([1, 64, 1060, 900]).cuda()
    b = torch.randn([1, 64, 530, 450]).cuda()
    c = torch.randn([1, 64, 265, 225]).cuda()
    model = PCD_Align().cuda()
    requires_grad(model, False)
    out = model([a, b, c], [a, b, c])
    print(out.shape)


if __name__ == '__main__':
    test_pcd()
```
and the result is correct:
![image](https://user-images.githubusercontent.com/18542006/129299760-20b6b06e-09c9-4030-88e1-a2953e6eb7ac.png)

**Comment by KeuntekLee** [comment]
Thanks for your reply.
This issue maybe related with dcn_v2 library.
I'm using pytorch 1.7 and dcn_v2_latest from other repository with merged PCD features.
I think need to to more experiments with environment which match to you noticed.
